frontend/packages/dev-console/src/utils/shared-submit-utils.ts

```diff
@@ -47,15 +48,18 @@ export const createService = (
   }
 
   const service: any = {
+    ...(oldService || {}),
     kind: 'Service',
     apiVersion: 'v1',
     metadata: {
+      ...(oldService ? oldService.metadata : {}),
       name,
       namespace,
-      labels: { ...defaultLabels, ...userLabels },
+      labels: { ...userLabels, ...defaultLabels },
```

Comment:
What's the reasoning for the order change?
I would think that a user's choice in labels should override our default labels.

Reply:
So to be able to edit labels from the form itself I populated the labels field with the existing labels but the issue was that if we are changing the image or app group for example then the labels gets a new value but since we already have these labels present if I spread the existing labels after the new labels then they get overridden. Probably we need to figure out a better way to handle this edge case.

Reply:
I think you'll have to have a smarter form and omit the labels that we are generating from the set of user labels. We shouldn't let the user see the `part-of` label for instance in the user labels field. That's controlled by the application field.